internal/manifests/collector/adapters/config_to_ports.go

```diff
@@ -58,3 +65,86 @@ func ConfigToMetricsPort(logger logr.Logger, config map[interface{}]interface{})
 
 	return int32(i64), nil
 }
+
+func GetServicePortsFromCWAgentOtelConfig(logger logr.Logger, config map[interface{}]interface{}) ([]corev1.ServicePort, error) {
+	ports, err := ConfigToComponentPorts(logger, ComponentTypeReceiver, config)
+	if err != nil {
+		logger.Error(err, "there was a problem while getting the ports from the receivers")
+		return nil, err
+	}
+
+	sort.Slice(ports, func(i, j int) bool {
+		return ports[i].Name < ports[j].Name
+	})
+
+	return ports, nil
+}
+
+// ConfigToComponentPorts converts the incoming configuration object into a set of service ports required by the exporters.
+func ConfigToComponentPorts(logger logr.Logger, cType ComponentType, config map[interface{}]interface{}) ([]corev1.ServicePort, error) {
+	// now, we gather which ports we might need to open
+	// for that, we get all the exporters and check their `endpoint` properties,
+	// extracting the port from it. The port name has to be a "DNS_LABEL", so, we try to make it follow the pattern:
+	// examples:
+	// ```yaml
+	// components:
+	//   componentexample:
+	//     endpoint: 0.0.0.0:12345
+	//   componentexample/settings:
+	//     endpoint: 0.0.0.0:12346
+	// in this case, we have 2 ports, named: "componentexample" and "componentexample-settings"
+	componentsProperty, ok := config[fmt.Sprintf("%ss", cType.String())]
+	if !ok {
+		return nil, fmt.Errorf("no %ss available as part of the configuration", cType)
+	}
+
+	components, ok := componentsProperty.(map[interface{}]interface{})
+	if !ok {
+		return nil, fmt.Errorf("%ss doesn't contain valid components", cType.String())
+	}
+
+	compEnabled := getEnabledComponents(config, cType)
+
+	if compEnabled == nil {
+		return nil, fmt.Errorf("no enabled %ss available as part of the configuration", cType)
```

Comment:
This doesn't seem like it should necessarily emit an error. It's valid to have a config with only scraping receivers.